shell/common/gin_helper/event_emitter_caller.h

```diff
@@ -64,6 +65,28 @@ v8::Local<v8::Value> CustomEmit(v8::Isolate* isolate,
                                       &converted_args);
 }
 
+template <typename T, typename... Args>
+v8::Local<v8::Value> CallMethodCatchException(v8::Isolate* isolate,
+                                              gin::Wrappable<T>* object,
+                                              const char* method_name,
+                                              Args&&... args) {
+  v8::EscapableHandleScope scope(isolate);
+  v8::Local<v8::Object> v8_object;
+  if (object->GetWrapper(isolate).ToLocal(&v8_object)) {
+    // Send unhandled errors within init to a different scope,
+    // so they're handled within JavaScript and don't crash within C++
+    v8::TryCatch try_catch(isolate);
+    v8::Local<v8::Value> value = scope.Escape(CustomEmit(
+        isolate, v8_object, method_name, std::forward<Args>(args)...));
+    if (try_catch.HasCaught()) {
+      node::errors::TriggerUncaughtException(isolate, try_catch);
```

Comment:
This is unsafe, we can't call this method in a `common` file.  Can we put it back in the callsite for now until we figure out a more generic strategy that supports non-node renderer contexts too.

cc @nornagon because I think you requested this change, we can move things to a helper but only once we've thought through the behavior in other contexts.  Let's fix the exact problem this time and refactor into a more generic solution at other callsites in the future

Reply:
just put it in a helper that's in `shell/browser`

Reply:
@nornagon @MarshallOfSound Converted the method into a static helper within web contents - that way, it's more DRY but stays close to the original call site. I added an overload since the two `_init` functions take slightly different params (one takes a gin handle, the other web_contents).

If we want to store the helper somewhere else or if there's another/better way to do the function overload, let me know, happy to change it! 🙂  Thank you both for reviewing too!

Reply:
Update: The helper was failing tests on Mac - I went ahead and moved the catches back to the original call sites.